### ISSUE TITLE
Add rule to check for inline-style links

### ISSUE THREAD

**Issue by pascalberger** [DavidAnson/markdownlint#40]
A rule to dissallow inline-style links would be a nice enhancements. 

Possible configuration parameters:
* Disallow all inline-style links
* Allow inline-style links in lists only
* Allow all inline-style links

**Comment by DavidAnson** [comment]
Can you please give some examples to be sure I understand the request? I searched "for inline-style" links and it suggests something like the following:

```html
<a href="..." style="something">...</a>
```

If so, does rule [MD033 Inline HTML](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md033---inline-html) help? It can be used to block HTML content which should cover the first of your three scenarios.

Also, why are lists special with regard to inline-style links? Would blockquotes deserve similar treatment? Tables? Just lists?

Thanks for the clarification!

**Comment by pascalberger** [comment]
No, I didn't mean inline HTML. Some examples to clarify:

Good:
```
Lorem ipsum dolor sit amet, [consectetur adipiscing] elit. Etiam lectus dui, laoreet eget mauris eget, interdum blandit risus. Vestibulum ac ornare dolor, et tincidunt nulla. Suspendisse vitae dolor nec ipsum vehicula varius non et diam. Vivamus finibus libero sed tincidunt accumsan. Aliquam erat volutpat. Donec faucibus, neque ac semper blandit, est lacus placerat purus, a blandit magna diam et lorem. Maecenas tincidunt iaculis eleifend. In bibendum elementum ex, sit amet pharetra est efficitur at. 

[consectetur adipiscing]: http://example.com
```

Bad, if not `Allow all inline-style links` is set:
```
Lorem ipsum dolor sit amet, [consectetur adipiscing](http://example.com) elit. Etiam lectus dui, laoreet eget mauris eget, interdum blandit risus. Vestibulum ac ornare dolor, et tincidunt nulla. Suspendisse vitae dolor nec ipsum vehicula varius non et diam. Vivamus finibus libero sed tincidunt accumsan. Aliquam erat volutpat. Donec faucibus, neque ac semper blandit, est lacus placerat purus, a blandit magna diam et lorem. Maecenas tincidunt iaculis eleifend. In bibendum elementum ex, sit amet pharetra est efficitur at. 
```

Good, if Àllow inline-style links in lists` is set:
```
* [Google](https://google.com)
* [Bing](https://bing.com)
```

The general idea behind the rule would be that inline-style links makes the Markdown source file harder to read. The exception for lists is that IMHO if you make a list of just links, it's easier to read if using inline-style than reference-style links, since the link is the whole content there. More precise this exception should only be valid if there is no additional text in the list, but don't know if this is possible :)

**Comment by DavidAnson** [comment]
Makes perfect sense, thank you!

**Comment by bhrutledge** [comment]
For completeness, an option to disallow footer links would be nice, and/or a "consistent" option, ala [ul-style](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md004).

**Comment by waldyrious** [comment]
Re-adding [comment](https://github.com/DavidAnson/markdownlint/issues/321#issuecomment-680123781) from #321, which was closed as a duplicate:

> It would also be nice if this was configurable by the length of the URL — i.e. non-reference links shorter than N characters would not trigger this rule, with the default max length being 0.
